compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -6419,6 +6422,7 @@ private void rewriteInvocation(BLangInvocation invocation, boolean async) {
             visitFunctionPointerInvocation(invocation);
             return;
         }
+        invocation.expr = isExprAndRequiredArgSame ? firstRequiredArg : invocation.expr;
         invocation.expr = rewriteExpr(invocation.expr);
```

Comment:
Shall we inline these two lines?

Reply:
Without initializing `firstRequiredArg`, can't we do, 
`invocation.expr = rewriteExpr(isExprAndRequiredArgSame ? invocation.requiredArgs.get(0) : invocation.expr)`

Reply:
`fixStreamTypeCastsInInvocationParams(invocation)` step changes the type of `requiredArgs`. We assigned the exact value to variable before change and used it